sqs/README.md

```diff
@@ -2,36 +2,237 @@
+
+For point-to-point scenarios, use the Channel Binding Object, as producers send to the queue and consumers receive from it directly.
+
+For publish-subscribe scenarios, use as a **publish** Operation Binding Object, as the producer sends to SNS and the consumer receives via SQS.
+
 <a name="version"></a>
 
 ## Version
 
 Current version is `0.1.0`.
 
-
 <a name="server"></a>
 
 ## Server Binding Object
 
 This object MUST NOT contain any properties. Its name is reserved for future use.
 
-
-
 <a name="channel"></a>
 
-## Channel Binding Object
-
-This object MUST NOT contain any properties. Its name is reserved for future use.
-
+## Channel Binding Object 
+
+Use the Channel Binding Operation for Point-to-Point SQS channels.
+
+There are three likely scenarios for use of the Channel Binding Object:
+
+- One file defines both publish and subscribe operations, for example if we were implementing the work queue pattern to offload work from an HTTP API endpoint to a worker process. In this case the channel would be defined on the Channel Object in that single file. 
+- The producer and consumer both have an AsyncAPI specification file, and the producer is raising an event, for example interop between microservices, and the producer 'owns' the channel definition and thus has the SQS Binding on its Channel Object. 
+- The producer and consumer both have an AsyncAPI specification file, and the consumer receives commands, for example interop between microservices, and the consumer 'owns' the channel  definition and thus has the SQS Binding on its Channel Object. 
+
+An SQS queue can set up a Dead Letter Queue as part of a Redelivery Policy. To support this requirement, the Channel Binding Object allows you to define both a Queue Object to use as the Channel or target in a *publish* Operation and a Dead Letter Queue. You can then refer to the Dead letter Queue in the Redrive Policy using the Identifier Object and setting the *name* field to match the *name* field of your Dead Letter Queue Object. (If you define the DLQ externally, the Identifier also supports an ARN). 
+
+### Fields
+|Field Name | Type | Description|
+|---|:---:|---|
+| <a name="channelBindingObjectQueue"></a>`queue` | [Queue](#queue)| **Required.** A definition of the queue that will be used as the channel. |
+| <a name="channelBindingObjectDLQ"></a>`deadLetterQueue` | [Queue](#queue)| **Optional.** A definition of the queue that will be used for un-processable messages. |
+|<a name="channelBindingObjectBindingVersion"></a>`bindingVersion` | string | **Optional**, defaults to `latest`. The version of this binding.|
+
+### Schemas
+
+#### Queue
+|Field Name | Type | Description|
+|---|:---:|---|
+|<a name="queueObjectRef"></a>$ref | `string` | Allows for an external definition of a queue. The referenced structure MUST be in the format of a [Queue](#queue). If there are conflicts between the referenced definition and this Queue's definition, the behavior is *undefined*.|
```

Comment:
Defined where? Is it a link, or is 'external' meaning it should go into `Components`? which does not define this behaviour

Reply:
I think we can remove the **queueObjectRef** because that looks like a **URI Reference** explained in [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986#section-4.1). This can be either **relative** or **absolute** reference.

URI References can be used for any object when it's needed. Any software libraries which serialise/deserialise JSON or YAML files must provide the option (or the default) to handle the references.

Because the **URI Reference** property (**$ref**) is a standard and can be used in any object I think it's unnecessary to specify the **$ref** property again in the **Queue** object.